browser/brave_search/brave_search_browsertest.cc

```diff
@@ -370,13 +372,36 @@ IN_PROC_BROWSER_TEST_F(BraveSearchTest, AdsStatusHeaderAdsDisabled) {
   run_loop.Run();
 }
 
+IN_PROC_BROWSER_TEST_F(BraveSearchTest, AdsStatusSearchAdsEnabled) {
+  base::RunLoop run_loop;
+  SetRequestExpectationsCallback(base::BindRepeating(
+      [](base::OnceClosure loop_closure,
+         const net::test_server::HttpRequest& request) {
+        const GURL url = request.GetURL();
+        EXPECT_FALSE(base::Contains(request.headers, kSearchAdsEnabledHeader));
+        if (request.GetURL().path_piece() == kBraveSearchPath) {
+          std::move(loop_closure).Run();
+        }
+      },
+      run_loop.QuitClosure()));
+
+  PrefService* prefs = browser()->profile()->GetPrefs();
+  prefs->SetBoolean(brave_rewards::prefs::kEnabled, true);
+  prefs->SetBoolean(brave_ads::prefs::kOptedInToSearchResultAds, true);
+
+  const GURL url = https_server()->GetURL(kAllowedDomain, kBraveSearchPath);
+  EXPECT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+
+  run_loop.Run();
+}
+
 IN_PROC_BROWSER_TEST_F(BraveSearchTest, AdsStatusHeaderNotAllowedDomain) {
```

Comment:
Maybe we should rename tests according to new header name?

Reply:
Renamed, but let me know if you have suggestions for better naming.

Reply:
new naming looks good for browser tests.